clang/lib/Driver/CMakeLists.txt

```diff
@@ -99,5 +100,6 @@ add_clang_library(clangDriver
   LINK_LIBS
   clangBasic
   clangLex
+  clangDependencyScanning
```

Comment:
As mentioned in my previous comment, this causes a cycle because clangDependencyScanning depends on clangDriver. This is fine when statically linking, but will fail with dynamic linking which we support.

This is only used by `DependencyScanningWorker::scanDependencies` to handle driver commands. To fix this I think it would be best to remove this dependency and instead have `scanDependencies` only work on frontend commands. I would then add a function to `clangDriver` that takes a `DependencyScanningWorker` and runs the scanner on all of frontend jobs, this would then be used to replace calls to `scanDependencies`.

This will need to be handled before this PR lands to avoid build issues.

Reply:
Frontend depends on Driver for the options (because driver and -cc1 options got merged) and for `clang::createInvocation`. I think we should move the options to their own library (clangOptions?). `clang::createInvocation` is harder because it's used by `ASTUnit::LoadFromCommandLine`. I think both of these should move to the driver as the rest of the frontend doesn't use them.